### PR TITLE
Add a binary crate to the autocompressor

### PR DESCRIPTION
**Pull request by Azrenbeth**
Builds on #67 

This allows compress_chunks_of_database to be triggered from the command line

### CODE REVIEW

File: auto_compressor/src/lib.rs [hunk 1]
```diff
@@ -6,5 +6,47 @@
...
+/// Helper struct for parsing the `default_levels` argument.
+///
+/// The compressor keeps track of a number of Levels, each of which have a maximum length,
+/// current length, and an optional current head (None if level is empty, Some if a head
+/// exists).
+///
+/// This is needed since FromStr cannot be implemented for structs
+/// that aren't defined in this scope
+#[derive(PartialEq, Debug)]
+pub struct LevelInfo(pub Vec<Level>);
```
Comment (erikjohnston) [inline auto_compressor/src/lib.rs:26]: Don't we already have a `LevelInfo` in the `synapse_compress_state` crate?

**Review by erikjohnston — APPROVED** [review]

File: auto_compressor/src/lib.rs (hunk 1 above)
Comment (Azrenbeth) [reply auto_compressor/src/lib.rs:26]: Nope! `continue_run` has an argument called `level_info` which has type `&[Level]` - maybe that's what you were thinking of?

Comment (erikjohnston) [reply auto_compressor/src/lib.rs:26]: Ah yeah, nevermind

Comment (Azrenbeth) [reply auto_compressor/src/lib.rs:26]: There is a `LevelSizes` struct that goes from comma seperated numbers to a vector of numbers though (which I guess I could reuse, but this way seems tidier)

**Pull request merged**